### PR TITLE
Help documentation

### PR DESCRIPTION
**Pull request by steevaavoo**

### CODE REVIEW

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #Requires -RunAsAdministrator
 function Get-sbRDSession {
     [OutputType('Custom.SB.RDSession')]
-    [cmdletbinding(DefaultParameterSetName = 'State')]
+    [cmdletbinding(DefaultParameterSetName = 'State', HelpUri = "https://bit.ly/2xghizR")]
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: I'd use full paths rather than redirect URLs here

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 2]
```diff
@@ -41,9 +41,11 @@ function Get-sbRDSession {
 
         if ($UserName) {
             Write-Verbose "Querying RD Session Collection for users like [$UserName]"
-            $sessions = Get-RDUserSession | Where-Object {
-                $_.UserName -like "*$UserName*"
-            }
+            $sessions = foreach ($user in $UserName) {
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1:89]: Good usage here; assigning foreach output :)

**Review by adamrushuk — CHANGES_REQUESTED** [review]
It's always good to include documentation, but I'm wondering where this information was gathered from, as I can't see proper comment-based help in any of the functions here: https://github.com/steevaavoo/ManageRDSessions/blob/help-documentation/ManageRDSessions/ManageRDSessions.psm1

Did you manually add some details after the Markdown files were produced?

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 1 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1]: Removed in favour of comment-based help

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 3]
```diff
@@ -1,14 +1,114 @@
 #Requires -RunAsAdministrator
+
 function Get-sbRDSession {
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1:3]: Make sure the CBH is aligned, and I'm not sure you should include the output within your examples (see my second example below:

```function Get-sbRDSession {
    <#
    .SYNOPSIS
        Get current sessions in a Remote Desktop Services deployment.
    .DESCRIPTION
        This cmdlet will return all Sessions in the current Remote Desktop Services deployment. Narrow results by user,
        or by a combination of minimum number of Idle Session minutes, Session State (Active, Disconnected, etc.) and choose
        whether to include yourself (the Admin user).
    .PARAMETER SessionState
        Returns only Remote Desktop sessions matching the specified State.
        Accepts: Active, Any, Connected, Disconnected or Idle.
    .PARAMETER IncludeSelf
        When enabled, this parameter will include the current (console) user within any matching search results. The
        current user is omitted by default as we don't normally want to send messages to/disconnect/log off ourselves.
    .PARAMETER MinimumIdleMins
        Specifies the minimum number of minutes for which each returned Remote Desktop should have been idle.
    .PARAMETER UserName
        Returns only Remote Desktop sessions matching the specified user name(s).
    .EXAMPLE
        Get-sbRDSession
        HostServer       : rds01.lab.milliondollar.me.uk
        UserName         : steve.baker
        UnifiedSessionID : 3
        SessionState     : STATE_ACTIVE
        IdleTime         : 0

        Getting all Remote Desktop sessions, regardless of status.
    .EXAMPLE
        Get-sbRDSession

        Getting all Remote Desktop sessions, regardless of status.        
    #>

    [OutputType('Custom.SB.RDSession')]
    ...
```

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 4]
```diff
@@ -1,14 +1,114 @@
...
-    [cmdletbinding(DefaultParameterSetName = 'State', HelpUri = "https://bit.ly/2xghizR")]
     param(
+        [cmdletbinding(DefaultParameterSetName = 'State')]
         [Parameter(Mandatory = $false, ParameterSetName = 'State')]
         [ValidateSet("Active", "Idle", "Connected", "Disconnected", "Any")]
         [Alias("State")]
         [string]$SessionState = "Any",
 
         # Switch to choose to include self - as we probably don't want to disconnect/logoff our own session, but
+
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1:56]: Blank line?

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 5]
```diff
@@ -83,6 +183,22 @@ function Get-sbRDSession {
 } #function
 
 function Disconnect-sbRDSession {
+    <#
+.SYNOPSIS
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: Copy my formatting example above (indentation etc)

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 6]
```diff
@@ -123,6 +239,23 @@ function Disconnect-sbRDSession {
 } #function
 
 function Remove-sbRDSession {
+    <#
+.SYNOPSIS
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: Copy my formatting example above (indentation etc)

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 7]
```diff
@@ -123,6 +239,23 @@ function Disconnect-sbRDSession {
...
+Remote Desktop session passed to it from the pipeline.
+
+.PARAMETER RDSession
+Requires an object passed by `Get-sbRDSession`.
+
+.PARAMETER AsJob
+Run the session log offs as background jobs, in parallel.
+
+.EXAMPLE
+PS C:\> PS C:\> Get-sbRDSession <parameters> | Remove-sbRDSession -AsJob
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: Dont include `PS C:\>` in the Examples

File: ManageRDSessions/ManageRDSessions.psm1 [hunk 8]
```diff
@@ -171,6 +304,30 @@ function Remove-sbRDSession {
 } #function
 
 function Send-sbRDMessage {
+    <#
+.SYNOPSIS
```
Comment (adamrushuk) [inline ManageRDSessions/ManageRDSessions.psm1]: Copy my formatting example above (indentation etc)

**Review by adamrushuk — CHANGES_REQUESTED** [review]
Much better now, just a few minor formatting issues really :)

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 3 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1:3]: Weirdly, the indentation locally matches the examples you've given below. I've indented the ending #> on each help block and re-pushed, let's see how it looks on GH

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 4 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1:56]: No idea why that was there. Gone :)

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 6 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1]: See above...

File: ManageRDSessions/ManageRDSessions.psm1 (hunk 8 above)
Comment (steevaavoo) [reply ManageRDSessions/ManageRDSessions.psm1]: See above...

**Review by adamrushuk — APPROVED** [review]

**Comment by adamrushuk** [comment]
BingPot! Looks good to me!

**Pull request merged**